src/Halcyon/HAL/HALAttributeResolver.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Halcyon.HAL.Attributes;
+
+namespace Halcyon.HAL
+{
+    internal class HALAttributeResolver
+    {
+        internal static void ResolveAttributes(HALResponse halResponse, object model)
+        {
+            var classAttributes = Attribute.GetCustomAttributes(model.GetType());
+            foreach (var attribute in classAttributes)
+            {
+                var modelAttribute = attribute as HalModelAttribute;
+                if (modelAttribute != null)
+                {
+                    var modelConfig = modelAttribute;
```

Comment:
What is the purpose of re-assigning the variable to a variable with a different name? I think the `modelAttribute` variable should be re-used.